indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
@drcrallen cool, I like it. I made the requested changes, let me know if there's anything else. Thanks!